ci-operator/step-registry/ipi/deprovision/deprovision/ipi-deprovision-deprovision-ref.yaml

```diff
@@ -7,5 +7,8 @@ ref:
     requests:
       cpu: 1000m
       memory: 300Mi
+  env:
+  - name: OS_CLIENT_CONFIG_FILE
```

Comment:
This is confusing. Why do you introduce a `OS_CLIENT_CONFIG_FILE` environment variable to later [redefine it with the `${SHARED_DIR}/` prefix](https://github.com/openshift/release/pull/14516/commits/90027e52efbdc4bbdb356fde12fd0d30c1d6f672#diff-eb60d72fa67a54162d11103fb4f56258cd241d4ec16dd549fc15b34449ca8a04R12)? Is it a filename or a path?
And why making the filename configurable when we [hardcode it to `clouds.yaml` anyway](https://github.com/openshift/release/pull/14516/commits/90027e52efbdc4bbdb356fde12fd0d30c1d6f672#diff-426cf0d05a84d593eaf64b40203a6b723a143f965c5e80d932eaa8b3a4ae1410R30)?

Reply:
**Why defining the env var in the step `ref`?**

The idea is to use the `env` property of the `ref` to declare dependencies. The `clouds.yaml` file from our self-managed secret is injected in `$SHARED_DIR` in the `conf` step.

**Why defining `OS_CLIENT_CONFIG_FILE` in `ref` and then redefining in the script?**

I originally defined `OS_CLIENT_CONFIG_FILE="${SHARED_DIR}/clouds.yaml"` in the `ref`. However you can't use variable substitution in the `ref` definition, so I ended up with rewriting it in the script.

**Do I think it's the only possible solution?**

No way. It's just one compromise I found acceptable; please help me find a better one.
One alternative solution is to drop the idea of using the `env` variables as documentation, and just hardcore everything.

Reply:
> don't inject clouds.yaml into SHARED_DIR, just use it directly from CLUSTER_PROFILE_DIR

What you propose is not possible with self-managed secrets unfortunately; `CLUSTER_PROFILE_DIR` can only contain secrets managed by DPTP.

Reply:
SHARED_DIR is safe for secrets; that’s why we use it for storing the install-config.